### ISSUE TITLE
Segfault caused while printing unicode character in interactive 'raku' session

### ISSUE THREAD

**Issue by chromis** [rakudo/rakudo#4054]
Hello,

I believe I found a basic interaction with the `raku` interpreter that causes (well, at least on my installation) a segmentation fault:

```
 % raku
Welcome to 𝐑𝐚𝐤𝐮𝐝𝐨™ v2020.11.
Implementing the 𝐑𝐚𝐤𝐮™ programming language v6.d.
Built on MoarVM version 2020.11.

To exit type 'exit' or '^D'
> sprintf "ux%04x = '%c'\n", $_, $_ for 0x2500 .. 0x25ff
Nil
> printf "ux%04x = '%c'\n", $_, $_ for 0x2500 .. 0x25ff
zsh: segmentation fault  raku
```

This interaction involves two lines... and executing the 2nd line that seems to cause the segfault without the 1st line "just works as expected":

```
 % raku 
Welcome to 𝐑𝐚𝐤𝐮𝐝𝐨™ v2020.11.
Implementing the 𝐑𝐚𝐤𝐮™ programming language v6.d.
Built on MoarVM version 2020.11.

To exit type 'exit' or '^D'
> printf "ux%04x = '%c'\n", $_, $_ for 0x2500 .. 0x25ff
ux2500 = '─'
ux2501 = '━'
ux2502 = '│'
ux2503 = '┃'
... and a whole lot more unicode characters :) ...
ux25fd = '◽'
ux25fe = '◾'
ux25ff = '◿'
> 
```

This is on...

```
 % raku -v
Welcome to Rakudo(tm) v2020.11.
Implementing the Raku(tm) programming language v6.d.
Built on MoarVM version 2020.11.

 % cat /etc/debian_version 
10.6

 % uname -a
Linux anomalo 4.19.0-12-amd64 #1 SMP Debian 4.19.152-1 (2020-10-18) x86_64 GNU/Linux
```

Thanks!

Regards,

Raymond.

**Comment by JJ** [comment]
This works in Ubuntu 2020.04, debian version /bullseye/sid,  uname
```
Linux princess-carolyn 5.4.0-53-generic #59-Ubuntu SMP Wed Oct 21 09:38:44 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```
Do you have something installed on top of the REPL? Like Readline or something like that?

**Comment by JJ** [comment]
Same problem with 2020.09

**Comment by JJ** [comment]
It works if it's assigned to something. Must be a problem with the sinking of sprintf. Also, it does not happen if the range is reduced

**Comment by chromis** [comment]
I tried to reproduce the issue while running raku in `valgrind` and for some reason I'm not able to reproduce it: `raku` runs very slow then, which is an indicator to me that `valgrind` "does something" ;) -- but no warnings or errors at all, unless I invoke `valgrind` in the wrong way which causes an incomplete analysis?

I sometimes saw such behavior when I needed to deal with concurrency-related issues.

**Comment by niner** [comment]
It segfaults in a `ctxouter` op:
```
MVM_spesh_frame_walker_move_outer (tc=0x406170, fw=0x7fffffffbe30) at src/spesh/frame_walker.c:263
263             MVMCode *code = (MVMCode *)fw->cur_caller_frame->work[i->code_ref_reg].o;
```
Apparently `fw->cur_caller_frame->work` is NULL

**Comment by chromis** [comment]
Hello,

It seems that this issue is fixed!

I tried to reproduce this on Rakudo v2021.10.

Perhaps due to the many improvements that were merged into 2021.10 (and mentioned in https://rakudoweekly.blog/2021/10/25/2021-43-thank-you/ ) ?

Can this be confirmed? If this is really fixed then this issue can be closed.

**Comment by niner** [comment]
Yes, it's quite probable that the issue has been fixed. As part of the new-disp work, we fixed several issues with the frame walker. I cannot reproduce it on the current version while I can easily reproduce the frame walker segfault on 2021.08. So I say this is gone for good!
